lib/pages/filter/model/filter.dart

```diff
@@ -60,6 +61,30 @@ class Filter {
     return list;
   }
 
+  List<FilterNode> findNodeByPath(List<String> path){
+    List<FilterNode> result = [root];
+    if(path == null){
+      return result;
+    }
+    for(int i = 0; i< path.length; i++){
+      FilterNode aux = _findNodeByPathHelper(path[i], result.last);
+      if(aux == null){
+        return result;
```

Comment:
Is this really what you want to do if aux is null? Doesn't that mean you didn't actually manage to find the node with the specified path?

Say we call `findNodeByPath(['BSc', 'CTI', 'ABC', 'DEF'])`. With the current implementation, the function returns the 'CTI' node and doesn't report an error.